test/functional/test_framework/util.py

```diff
@@ -323,12 +330,13 @@ def get_auth_cookie(datadir):
                 if line.startswith("rpcpassword="):
                     assert password is None  # Ensure that there is only one rpcpassword line
                     password = line.split("=")[1].strip("\n")
-    if os.path.isfile(os.path.join(datadir, "regtest", ".cookie")) and os.access(os.path.join(datadir, "regtest", ".cookie"), os.R_OK):
-        with open(os.path.join(datadir, "regtest", ".cookie"), 'r', encoding="ascii") as f:
-            userpass = f.read()
-            split_userpass = userpass.split(':')
-            user = split_userpass[0]
-            password = split_userpass[1]
+    cookie = os.path.join(datadir, "regtest", ".cookie")
+    wait_until(lambda: can_read_file(cookie), timeout=5)
```

Comment:
I believe this method is not allowed to block, it must return a result immediately

Can be fixed by keeping the previous `if`-logic

Reply:
> I believe this method is not allowed to block, it must return a result immediately

Why not?

Reply:
See `wait_for_rpc_connection`, which calls this method

Reply:
Thanks, `wait_for_rpc_connection` already retries so I've changed this remove the early checks and just try opening the cookie file.